server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -236,6 +243,7 @@ public Object accumulate(Object accumulated, Object in)
 
       try {
         final Query theQuery = query;
+        final QueryToolChest theToolChest = toolChest;
```

Comment:
If I'm reading this correct, the only reason you need to have a 2nd variable name here is because the two code paths of A) query successfully deserializing and B) query throws exception while deserializing... are all muddled together in the exception handling.

Would it be worth it in this PR to try and clean that up a bit?